fedimint-rocksdb/src/lib.rs

```diff
@@ -150,6 +150,19 @@ impl<'a> IDatabaseTransaction<'a> for RocksDbTransaction<'a> {
         self.0.commit()?;
         Ok(())
     }
+
+    fn rollback(&mut self) {
+        match self.0.rollback_to_savepoint() {
+            Ok(()) => {}
+            _ => {
+                warn!("Rolling back database transaction without a set savepoint");
+            }
+        }
+    }
+
+    fn set_tx_savepoint(&mut self) {
```

Comment:
Do we need it?

Reply:
I think its worth exposing in the trait. It allows us to support more complex database transactions, such as saving a transaction to an intermediate state, then rolling back to that if something later on fails (example in test_basic_dbtx_rw).

Reply:
As discussed offline, we will keep this in the trait for now. If we find that it is not needed at all in fedimintd, we can remove it from the trait.